### PR TITLE
Update design - use Sveltekit

### PR DESCRIPTION
**Pull request by Visual-Dawg**
Use Sveltekit to generate the site.
Also worked a bit on the design and stuff

### CODE REVIEW

**Comment by LeoDog896** [comment]
Hello!

The author isn't doing any PRs that introduce "[unnecessary refactors](https://github.com/hyprwm/hyprland-website/issues/17#issuecomment-1412248460)" as the author doesn't want to use frameworks in their website (See https://github.com/hyprwm/hyprland-website/issues/17) in order to avoid [bikeshedding](https://en.wikipedia.org/wiki/Law_of_triviality) and arguments, nor are they accepting any deployment strategy changes (I noticed you're using [@sveltejs/adapter-static](https://github.com/hyprwm/hyprland-website/pull/28/files#diff-6317e218ea40a2a9b47af98e100e81c06f9c0abff66737e6ca06ec4b29402242), but the author also doesn't want any of these changes either (https://github.com/hyprwm/hyprland-website/issues/24).

(Out of separate curiosity, however, what are these design change?)

**Comment by zelfroster** [comment]
Dude, At least ask the devs before working on a big PR. Or look at old issues for a similar one.
There are already a few requests for using a Framework or SSG, but the owner is not open to ideas, Even I agree to his opinion, a simple website should not be made more complicated than it already is. Pure HTML + CSS + maybe some JS is more than enough for a simple website.

Also What are those commit messages? Looks like troll commits

**Comment by zelfroster** [comment]
> Dude, At least ask the devs before working on a big PR. Or look at old issues for a similar one. There are already a few requests for using a Framework or SSG, but the owner is not open to ideas, Even I agree to his opinion, a simple website should not be made more complicated than it already is. Pure HTML + CSS + maybe some JS is more than enough for a simple website.
> 
> Also What are those commit messages? ~~Looks like troll commits~~ Improve them next time when you work, make them more clear.

**Comment by Visual-Dawg** [comment]
Hey guys, I talked with Vax, its all good. 
Yes, the commits are a bit messy, but its actually a pretty new website and they should get squashed anyway.
I showed the progress on the Discord

**Comment by NotAShelf** [comment]
P.S. you don't have to be smartasses

**Comment by LeoDog896** [comment]
> Hey guys, I talked with Vax, its all good. Yes, the commits are a bit messy, but its actually a pretty new website and they should get squashed anyway. I showed the progress on the Discord

Awesome! Are you planning to use GitHub actions for deployment, or are y'all continuing with the normal run on the server sort of situation?

**Comment by zelfroster** [comment]
> Hey guys, I talked with Vax, its all good. Yes, the commits are a bit messy, but its actually a pretty new website and they should get squashed anyway. I showed the progress on the Discord

Didn't knew What's going on behind the scenes. I am actually surprised that Vaxry agreed for using anything else than simple stuff.

**Comment by vaxerski** [comment]
I got ~~bribed~~ convinced to allow this

**Comment by NotAShelf** [comment]
$20 is $20

**Comment by vaxerski** [comment]
image

footer is fucked, margin on the bottom?

discord section - to be removed imo (especially since it might disappear soon)

image

"other options"

image

these animations here are annoying tbh, I am not a fan

Major note: the host I use cannot use pnpm. only npm.

**Comment by Visual-Dawg** [comment]
Thank you for the feedback.
I will change those. 

The site is currently set to be build as a purely static site, so a host server is not nessecary. Something like GithubPages is already enough. 
I can change that though. And running the commands with their `npm` counterpart will have the same results in this case.

**Comment by vaxerski** [comment]
ok then, no need to change shit if it's static

**Comment by Visual-Dawg** [comment]
Added the changes. I can also take a look on how to use Github Pages with that, if you want

**Comment by Vagahbond** [comment]
super cool looking :+1:

**Comment by Visual-Dawg** [comment]
Note to self: Remove no-index when finished

**Comment by Visual-Dawg** [comment]
This is the current status:
http://hyprtesty.surge.sh

I think its ready

**Comment by vaxerski** [comment]
> I think its ready

and then man pushes like 30 commits

decide

**Comment by vaxerski** [comment]
I told your ass to make this scroll and not redirect to wiki:
image

image

Check out

image

why does the HoF only contain the oldest rices? Why not random / winner of each competition?

image

why is this gibberish japanese in the back here

**Comment by Visual-Dawg** [comment]
> and then man pushes like 30 commits
I received some more feedback from outfoxxed, which I wanted to include.

> I told your ass to make this scroll and not redirect to wiki:
The navbar will be visible on all pages, so it cant scroll down unless its the home page.
In general, I wanted this to be consistent. (All install buttons lead to the wiki)

> Check out
Ill fix this in a sec

> why does the HoF only contain the oldest rices? Why not random / winner of each competition?
Because their colors create the nicest perspective. Warmer colors are closer, coolers more far away. Here flick0s rice is perfect to be in front and then the last one has a lot of blue and blends nicely with the blue to create a sense of depth / atmosphere.
I played around with the new rices, but they are mostly black and it just doesnt look that nice.

> why is this gibberish japanese in the back here
It looks cool

**Comment by vaxerski** [comment]
you can use this for japanese https://www.reddit.com/r/beemovie/comments/ge52xx/bee_movie_script_in_japanese/

**Comment by Visual-Dawg** [comment]
For whatever reason I cannot update the site on surge.sh, but I did change the japanese text with a part of the linked post

**Comment by vaxerski** [comment]
brilliant, it's the bee movie script in japanese

**Comment by Visual-Dawg** [comment]
It also looks better now 👍 
If there isnt anything else, this can get merged

File: src/routes/CommunitySlice.svelte [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+			target="_blank"
+			rel="noopener"
+			aria-label="Join us on Discord"
+		>
+			<DiscordIcon class="h-full w-full  " />
+		</a>
+
+		<a href="https://discord.com/invite/hQ9XvMUjjr">
+			<Button type="fancyOutline">Join us on Discord</Button>
+		</a>
```
Comment (NotAShelf) [inline src/routes/CommunitySlice.svelte:233]: I'm not familiar with Sveltekit, but in Next we usually have routes such as `domain.tld/discord` that we can then use inside the code to reduce redundancy and improve maintainability. If it's possible (and I'm guessing it is), I'll recommend poiting a single `hyprland.org/discord` to the discord invite and using that route in all discord links

File: src/routes/FeaturesSlice.svelte [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+						alt=""
+						aria-hidden="true"
+						loading="lazy"
+					/>
+				</div>
+			</div>
+		</FeatureCard>
+		<FeatureCard class="" title="Dynamic tiling" color="purple">
+			<p class="max-w-[60ch]">
+				Automatic tiling which just works. Supports multiple fine-tuneable layouts.
```
Comment (NotAShelf) [inline src/routes/FeaturesSlice.svelte:114]: Automatic tiling *that* just works.

File: src/routes/hall_of_fame/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+		/>
+		<FamedRice
+			name="Day and Night"
+			creator="Mathisbuilder"
+			pretitel="#3"
+			dotfilesLink="https://github.com/MathisP75/summer-day-and-night"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/98901170?v=4"
+			thumbnail="/imgs/ricing_competitions/2/day-night.webp"
+		/>
+	</Contest>
```
Comment (NotAShelf) [inline src/routes/hall_of_fame/+page.svelte:77]: I'd use a function that iterates over a json file to construct the FamedRice components here

File: src/routes/hall_of_fame/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+		<FamedRice
+			name="Unnamed"
+			creator="lauroro"
+			pretitel="#3 (ex aequo)"
+			dotfilesLink="https://github.com/lauroro/hyprland-dotfiles"
+			creatorProfilePicture="https://avatars.githubusercontent.com/u/88981092?s=48&v=4"
+			thumbnail="/imgs/ricing_competitions/1/lauroro.webp"
+			class="pb-16"
+		/>
+	</Contest>
```
Comment (NotAShelf) [inline src/routes/hall_of_fame/+page.svelte:121]: see above

**Review by NotAShelf — COMMENTED** [review]
LGTM

I've got some minor nitpicks, and other changes for improved maintainability - none of them are dealbreakers, but would be appreciated

File: src/routes/hall_of_fame/+page.svelte (hunk 4 above)
Comment (Visual-Dawg) [reply src/routes/hall_of_fame/+page.svelte:121]: Good points, Ill update the code

Comment (Visual-Dawg) [reply src/routes/hall_of_fame/+page.svelte]: Code is updated :)

**Comment by vaxerski** [comment]
push it to hyprtesty and lmk

**Comment by Visual-Dawg** [comment]
[ hyprtesty.surge.sh](http://hyprtesty.surge.sh/)
Aight, here you go

**Comment by vaxerski** [comment]
all the easter egg tiles on the main page are laggy / buggy if you quickly hover off of them. No animations, they snap into place.

**Comment by Visual-Dawg** [comment]
> all the easter egg tiles on the main page are laggy / buggy if you quickly hover off of them. No animations, they snap into place.

Damn, Firefox at it again. Ill see if I can work around that

**Comment by NotAShelf** [comment]
also seeing minor frame drops on main page videos

**Comment by Visual-Dawg** [comment]
Ahhhh, Ill probably the newish codecs. Much smiller videos, but takes more power to play.
Maybe Ill find a middle ground

**Comment by Visual-Dawg** [comment]
Aight, Im using h264 again, which plays back smoothly and added a nice slow fade to the background images in the hero section.

I also re-encoded end_4's rice, so that the file can start playback without being fully downloaded and as I received feedback that the slide-on-hover in the plugins can be confusing (while I also received feedback that its weird if it does not slide) - I added a button which on click slides the video in.

Also, if there are some small optimizations here, I would prefer that the site still gets published/merged and that those then become issues, which I can than do in batches.

**Review by NotAShelf — APPROVED** [review]
looks good to go

**Comment by vaxerski** [comment]
the tiles still snap back if I unhover while they are enlarged

**Comment by Visual-Dawg** [comment]
@vaxerski Ah, I misunderstood you. I thought you were talking about the background-image.
Havnt spotted this one. 
Ill fix this one

**Comment by Visual-Dawg** [comment]
Aight, I updated it

**Review by vaxerski — APPROVED** [review]
lgtm

**Comment by vaxerski** [comment]
thanks!

**Pull request merged**